routes/truckRoutes.js

```diff
@@ -1,3 +1,70 @@
-// connect to mongodb in this module as this is where you'll be making creat/read/delete calls to your database
+// connect to mongodb in this module as this is where you'll be making create/read/delete calls to your database
 // use 'mongodb://localhost/foodTruckAPI' for your mongoose connection string
 // remember this is a Node module
+
+var express = require('express');
+
+// the Router method returns an instance which can be mounted as middle-ware
+var router = express.Router();
+
+// '/trucks' routes
+app.route('/trucks')
+
+	// List all trucks
+	.get(function (request, response) {
+		Truck.find(function (error, trucks) {
+			if (error) {
+				response.status(500).send(error);
+			} else {
+				response.json(trucks);
+			}
+		});
+	})
+
+	// Add new truck
+	.post(function (request, response) {
+		var truck = new Truck(request.body);
+
+		truck.save(function (error) {
+			if (error) {
+				response.status(500).send(error);
+			} else {
+				response.status(201).send(truck);
+			}
+		});
+	});
+
+
+// '/trucks/truckId' routes
+app.route('/trucks/truckId')
+
+	// Retrieve a truck object
+	.post(function (request, response) {

```

Comment:
You don't want to define a post handler for this route. Using post is for adding a new food truck, and this route already has an id, so it would be either get, put, or delete on an existing truck.